File: apkEditor/src/main/java/com/mcal/apkeditor/ComposeThread.java
package com.mcal.apkeditor;

import com.mcal.apkeditor.ce.IApkMaking;
import com.mcal.common.utils.ITaskCallback;

import java.util.List;
import java.util.Map;
import java.util.Set;

public abstract class ComposeThread extends Thread {
    public abstract void stopRunning();

    public abstract void setExtraMaker(IApkMaking extraMaker);

    public abstract void setModification(boolean strModified, boolean manifestModified,
                                         boolean resFileModified, List<String> modifiedSmaliFolders,
                                         Map<String, String> addedFiles, Map<String, String> replacedFiles,
                                         Set<String> deletedFiles, Map<String, String> fileEntry2ZipEntry,
                                         boolean bSignApk);

    public abstract void setTaskCallback(ITaskCallback callback);
}

File: apkEditor/src/main/java/com/mcal/apkeditor/ComposeThread.kt
package com.mcal.apkeditor

import com.mcal.apkeditor.ce.IApkMaking
import com.mcal.common.utils.ITaskCallback
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Job

abstract class ComposeThread : CoroutineScope {
    abstract fun stopRunning()
    abstract fun setExtraMaker(extraMaker: IApkMaking?)
    abstract fun setModification(
        strModified: Boolean,
        manifestModified: Boolean,
        resFileModified: Boolean,
        modifiedSmaliFolders: List<String>?,
        addedFiles: MutableMap<String, String>?,
        replacedFiles: MutableMap<String, String>?,
        deletedFiles: MutableSet<String>?,
        fileEntry2ZipEntry: Map<String, String>?,
        bSignApk: Boolean
    )

    abstract fun setTaskCallback(callback: ITaskCallback?)

    abstract fun execute(): Job
}